Trippy/Utils/TSPUtils.m

```diff
@@ -0,0 +1,95 @@
+//
+//  TSPUtils.m
+//  Trippy
+//
+//  Created by Catherine Lu on 7/19/22.
+//
+
+#import "TSPUtils.h"
+#import "Itinerary.h"
+
+@implementation TSPUtils
+
++ (NSArray *)permutations:(NSArray *)elements {
+    if (elements.count <= 1) {
+        return @[elements];
+    }
+    NSArray *subarray = [elements subarrayWithRange:NSMakeRange(1, elements.count-1)];
+    NSMutableArray *results = [[NSMutableArray alloc] init];
+    for (NSArray *res in [self permutations:subarray]) {
+        for (int i = 0; i < elements.count; i+=1) {
+            NSArray *firstHalf = [res subarrayWithRange:NSMakeRange(0, i)];
+            NSArray *current = [elements subarrayWithRange:NSMakeRange(0, 1)];
+            NSArray *secondHalf = [res subarrayWithRange:NSMakeRange(i, res.count-i)];
+            NSArray *res = [[firstHalf arrayByAddingObjectsFromArray:current] arrayByAddingObjectsFromArray:secondHalf];
+            [results addObjectsFromArray:@[res]];
+        }
+    }
+    return results;
+}
+
++ (NSArray *)reorder:(NSArray *)elements order:(NSArray *)order {
+    NSMutableArray *newList = [[NSMutableArray alloc] init];
+    for (NSNumber *ix in order) {
+        [newList addObject:[elements objectAtIndex:[ix intValue]]];
+    }
+    return newList;
+}
+
++ (int)totalDistance:(NSArray *)order matrix:(NSDictionary *)matrix {
+    int sum = 0;
+    int firstIndex = [[order firstObject] intValue];
+    NSNumber *initialWeight = matrix[@"rows"][0][@"elements"][firstIndex+1][@"distance"][@"value"];
+    sum += [initialWeight intValue];
+    for (int i=1; i < order.count; i+=1) {
```

Comment:
`i+=1`... I'm tickled by this. `i++` is more conventional. Is this ported python code? 😄

Reply:
Oops yes, forgot that i++ was a thing, switched it out!